### PR TITLE
Escaping - Part 1

### PR DESCRIPTION
**Pull request by hoseinrafiei**
## Problem this Pull Request solves
It's a starter for https://github.com/eventespresso/event-espresso-core/issues/3738 issue.

## How has this been tested
I did full E2E tests for the upcoming event widget in both of classic editor and block editor.

Also I changed some codes on a shortcode for thank you page which I think is deprecated and have no use at the moment.

## Screenshots for Tests

Screen Shot 2022-02-23 at 18 01 25

Screen Shot 2022-02-23 at 18 00 52

### CODE REVIEW

**Comment by hoseinrafiei** [comment]
@tn3rb 

I investigated the ticket and there are lots of jobs to be done for this so I split the task to following 2 phases.

### Escaping Phase
I found about 800 variables that should be checked for escaping in following directories. I will check and fix them part by part and will create separate PRs for them.

admin_pages (188)
caffeinated (330)
core (120)
modules (110)
public (19)
shortcodes (5)
widgets (21)

Other directories are ignored either due to development purposes or zero variables to get escaped.

### wp_kses Phase
After the first phase, I will start checking the remained variables to apply `wp_kses` function. Right now there are lots of variables that should be checked but after doing the first phase, there will be much less to for investigation.

I will improve the `core/services/request/sanitizers/AllowedTags.php` since there are many more tags and attributes that should be added. I may add some functions to get the tags based on context as you suggested.

Please let me know your thoughts for this PR and also for the whole plan. Your feedback will be a great help.

File: shortcodes/espresso_thank_you/templates/thank-you-page-transaction-details.template.php [hunk 1]
```diff
@@ -44,7 +42,7 @@
                     <?php $transaction->e_pretty_status(true);
                     if ($show_try_pay_again_link && ! $transaction->is_completed()) { ?>
                         <span class="small-text">
-                        <a href='<?php echo $SPCO_payment_options_url ?>'>
+                        <a href='<?php echo esc_url($SPCO_payment_options_url); ?>'>
```
Comment (tn3rb) [inline shortcodes/espresso_thank_you/templates/thank-you-page-transaction-details.template.php:45]: `esc_url()` is fine for display, but for URLs that need to work, it's better to use `esc_url_raw()`

File: shortcodes/espresso_thank_you/templates/thank-you-page-transaction-details.template.php [hunk 2]
```diff
@@ -68,7 +66,7 @@
 
     <?php if ($show_try_pay_again_link && ! $transaction->is_completed()) { ?>
         <p class="small-text jst-rght">
-            <a href='<?php echo $SPCO_payment_options_url ?>'><?php
+            <a href='<?php echo esc_url($SPCO_payment_options_url); ?>'><?php
```
Comment (tn3rb) [inline shortcodes/espresso_thank_you/templates/thank-you-page-transaction-details.template.php:69]:
```suggestion
            <a href='<?php echo esc_url_raw($SPCO_payment_options_url); ?>'><?php
```

File: widgets/upcoming_events/EEW_Upcoming_Events.widget.php [hunk 3]
```diff
@@ -357,7 +357,7 @@ public function make_the_title_a_link($title)
      */
     public function fieldID($field_name)
     {
-        return esc_attr(parent::get_field_id($field_name));
+        return parent::get_field_id($field_name);
```
Comment (tn3rb) [inline widgets/upcoming_events/EEW_Upcoming_Events.widget.php:360]: 👍🏻

**Review by tn3rb — CHANGES_REQUESTED** [review]
looks great but need to swap out of of the escape functions being used

**Comment by hoseinrafiei** [comment]
> looks great but need to swap out of of the escape functions being used

Thanks @brent. I fixed those items, Please check it again.

**Review by tn3rb — APPROVED** [review]
yup looks great now, thnx 💯

**Comment by hoseinrafiei** [comment]
@mohdshakeel Please check this branch and do an E2E test.
After that please inform me to merge the branch.

**Comment by mohdshakeel** [comment]
I have tested the widget **Event Espresso Upcoming Events** . I picked each settings options one by one and nothing found any issue with its back-end settings and front-end display.
Front end is displaying perfect with each setting of widget.

image

**Review by mohdshakeel — COMMENTED** [review]
Can We use the **wp_get_attachment_image** function at **public/Espresso_Arabica_2014/content-espresso_events-thumbnail.php** to display the feature image? It may better in terms of speed optimization.

**Comment by tn3rb** [comment]
> Can We use the wp_get_attachment_image function at public/Espresso_Arabica_2014/content-espresso_events-thumbnail.php to display the feature image? It may better in terms of speed optimization.

that would be addressed in a separate issue, but is not important atm as we will be migrating everything over to blocks at some point

**Pull request merged**